packages/react-components/react-tree/src/components/TreeItem/useTreeItemContextValues.ts

```diff
@@ -1,27 +1,26 @@
-import type { TreeItemContextValue } from '../../contexts';
+import * as React from 'react';
 import type { TreeItemContextValues, TreeItemState } from './TreeItem.types';
+import type { TreeItemContextValue, TreeItemSlotsContextValue } from '../../contexts';
 
-export function useTreeItemContextValues_unstable(
-  state: Pick<TreeItemState, keyof TreeItemContextValue>,
-): TreeItemContextValues {
-  const { value, isActionsVisible, isAsideVisible, actionsRef, itemType, layoutRef, subtreeRef, expandIconRef, open } =
-    state;
+export function useTreeItemContextValues_unstable(state: TreeItemState): TreeItemContextValues {
+  const { value, itemType, layoutRef, subtreeRef, open, actions, aside, expandIcon } = state;
 
   /**
    * This context is created with "@fluentui/react-context-selector",
    * there is no sense to memoize it
    */
   const treeItem: TreeItemContextValue = {
-    isActionsVisible,
-    isAsideVisible,
     value,
-    actionsRef,
     itemType,
     layoutRef,
     subtreeRef,
-    expandIconRef,
     open,
   };
 
-  return { treeItem };
+  const treeItemSlots: TreeItemSlotsContextValue = React.useMemo(
+    () => ({ actions, aside, expandIcon }),
+    [actions, aside, expandIcon],
+  );
```

Comment:
This memo is kinda useless as it will trigger updates once one of these changes. Probably it's better idea to have a separate context for each slot.

Reply:
I don't think it would make any difference, as the consumer to this context would be the same for the 3 slot values, so if any of them updates a re-render is inevitable anyway.

Perhaps one good thing to do would be to memoize the slot creation though 🤔

Reply:
Let's keep it as is for now then.